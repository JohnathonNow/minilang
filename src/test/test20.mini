fun fact(N) do
	var F := 1
	for I in 1 .. N do
		F := old * I
	end
	ret F
end

print('20! = {fact(20)}\n')

let Start := now()
for I in 1 .. 1000000 do
	fact(20)
end
print('Time taken = {now() - Start}\n')
