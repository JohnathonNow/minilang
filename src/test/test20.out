test(nil) -> <nil>
test(hello) -> <string or list>
test([]) -> <string or list>
test(1) -> <unknown>
test(50) -> 100 50, or > 200
test(100) -> 100 50, or > 200
test(200) -> <unknown>
test(201) -> 100 50, or > 200
test(65) -> 60 .. 70
